source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. tip::
+  
+   Only use ``--sandbox`` and ``--writable`` when developing the image. Once the build is settled, create the container with a definitions file and distribute it as-is for maximum stability.
+
+There is also a full HPC development image made available by Nvidia, called "HPC SDK", which is the software development kit that contains all the compilers, libraries and tools necessary to build efficient code that runs on GPUs. This image can be found `here <Https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nvhpc>`_.
+
+Running python
+==============
+
+Popular python interfaces for modelling are tensorflow, keras, pytorch, and more. An example for using tensorflow in singularity is provided below, but some warnings have to be taken into account, due to the default behaviour of singularity with the host machine. 
+
+Starting on a machine in the GPU partition, we create a definitions file ``nv-tf-22.07.def`` containing:
+
+.. code-block:: bash
+
+  Bootstrap: docker
+  From: nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+  %post
+  cd /tmp
+  git clone https://github.com/tensorflow/docs
+ 
+  %runscript
+  cd /tmp/docs/site/en/tutorials/keras
+  python
+ 
+  %help
+  This is a demo container to show how to run tensorflow in python
+
+and build the container using the usual 
+
+.. code-block:: bash
+
+   singularity build --nv --fakeroot nv-tf-22.07.sif nv-tf-22.07.def
+
+In this definitions file, the tensorflow docs and tutorials are installed as an example to show how to do it. 
+
+.. WARNING::
+   Running ``pip`` inside the container using ``singularity shell`` when it is in ``--writable`` mode will write the python libraries to the default **mounted** location. This location is the ``$HOME``-folder of ``$USER``. As such, pip packages will end up on the host machine and not in the container. To avoid this behaviour, only run ``pip`` during the building of the image in the definitions file, or change the mounting behaviour of singularity when entering the shell. For example, mount the local path of your project as working directory as the ``$HOME`` in the container. 
+
+   For information on this, read ``man singularity-shell`` and `bind mounts <https://singularity-userdoc.readthedocs.io/en/latest/bind_paths_and_mounts.html>`_.
+
+.. WARNING::
+   As the home folder is mounted by default in singularity, and python searches certain folders by default, it is possible that inside the container packages from the host machine are called, instead of what is inside the container. For example, the ``~/.local`` folder on the host machine can have presedence over site-packages in the container. If errors appear relating to CUDA ``.so`` files, or versions of packages are mismatching, ensure that the user-space is not accidentally providing libraries to the container.
```

Comment:
1) typo: precedence
2) how to prevent that user-space is accidentally providing libs to cont i.e as a user what should i do?